ENIGMAsystem/SHELL/Widget_Systems/xlib/dialogs.cpp

```diff
@@ -18,17 +18,28 @@
 #include "dialogs.h"
 #include "Widget_Systems/widgets_mandatory.h"
 #include "Widget_Systems/General/WSdialogs.h"
+#include <X11/Xlib.h>
 #include <string>
 using std::string;
 
 namespace enigma {
 
+CommandLineWidgetEngine *current_widget_engine = zenity_widgets;
+
+static bool kwin_running() {
+  Display *d = XOpenDisplay(NULL);
+  Atom aKWinRunning = XInternAtom(d, "KWIN_RUNNING", True);
+  bool bKWinRunning = (aKWinRunning != None);
+  XCloseDisplay(d);
+  return bKWinRunning;
+}
+
 bool widget_system_initialize() {
+  // Defaults to the GUI toolkit (GTK+/Qt) that matches Desktop Environment.
+  current_widget_engine = kwin_running() ? kdialog_widgets : zenity_widgets;
```

Comment:
KDE running doesn't really guarantee that kdialog is available. If both Zenity and KDialog are available, then you should of course prefer whichever system is running, but odds are, the user has neither one installed. And if they do, they almost certainly only have one or the other. That said, this is a better guess than what we had before.

Reply:
![20190821_184542](https://user-images.githubusercontent.com/4379204/63473640-4078f580-c444-11e9-90b4-796957f7c729.jpg)